src/main/java/edu/wpi/grip/core/operations/composite/FindContourOperation.java

```diff
@@ -0,0 +1,96 @@
+package edu.wpi.grip.core.operations.composite;
+
+import com.google.common.eventbus.EventBus;
+import edu.wpi.grip.core.InputSocket;
+import edu.wpi.grip.core.Operation;
+import edu.wpi.grip.core.OutputSocket;
+import edu.wpi.grip.core.SocketHint;
+
+import java.io.InputStream;
+import java.util.Optional;
+
+import static org.bytedeco.javacpp.opencv_core.*;
+import static org.bytedeco.javacpp.opencv_imgproc.*;
+
+/**
+ * An {@link Operation} that, given a binary image, produces a list of contours of all of the shapes in the image
+ */
+public class FindContourOperation implements Operation {
+
+    public final static class Contours {
+        protected int rows = -1, cols = -1;
+        protected MatVector contours = new MatVector();
+
```

Comment:
> Give this class a constructor like any normal object.

The constructor would be empty, which is the same as the implicit constructor.

> Let the java memory model clean it up like any normal object.

What am I doing to stop that?

> If you don't want to keep allocating MatVectors then use the operations, createData() method to allocate a single MatVector to use as the output. The rest of these Contour objects can be flushed (ie. deleted) by the jvm.

I don't care about allocating MatVectors.  They don't take up a lot of space.